GitHub App: add docs (#12114)
readthedocs/readthedocs.org: docs/user/reference/git-integration.rst

How automatic configuration works
---------------------------------

> When you Read the Docs account is connected to GitHub, and you :doc:`add a new Read the Docs project </intro/add-project>`:

Reviewer: ```suggestion
When your Read the Docs account is connected to GitHub, and you :doc:`add a new Read the Docs project </intro/add-project>`:
```

* Read the Docs automatically connects your project with the GitHub repository,
  and subscribes to the repository's events.
* Read the Docs makes use of its :ref:`GitHub App <reference/git-integration:GitHub App>` to interact with your repository.

When your Read the Docs account is connected to GitLab or Bitbucket, and you :doc:`add a new Read the Docs project </intro/add-project>`:

* Read the Docs automatically creates a Read the Docs Integration that matches your Git provider.
* Read the Docs creates an incoming webhook with your Git provider, which is automatically added to your Git repository's settings using the account connection.

> * Read the Docs creates an SSH key, which is automatically added to your Git repository (when importing private repositories on |com_brand|).

Reviewer: ```suggestion
* Read the Docs creates an SSH key, which is automatically added to your Git repository (when adding private repositories on |com_brand|).
```

You are asked to grant permissions for Read the Docs to perform a number of actions on your behalf.

At the same time, we use this process for authentication (login)
since we trust that the user who connects the account is the owner of Git provider account.

By granting Read the Docs the requested permissions,
we are issued a secret OAuth token from your Git provider.

> In the case of GitLab and Bitbucket, we can use the secret token
> to automatically configure repositories during :doc:`project creation </intro/add-project>`,
> for GitHub, you need to install our :ref:`GitHub App <reference/git-integration:GitHub App>` in the repository you want to import.

Reviewer: ```suggestion
In the case of GitLab and Bitbucket, we can use the secret token
to automatically configure repositories during :doc:`project creation </intro/add-project>`.

On the other hand, for GitHub, you need to install our :ref:`GitHub App <reference/git-integration:GitHub App>` in the repository you want to import.
```

      When installing the Read the Docs GitHub App in a repository, you will be asked to grant the following permissions:

      Repository permissions
        Commit statuses (read and write)
          This allows Read the Docs to report the status of the build to GitHub.
        Contents (read only)
          This allows Read the Docs to clone the repository and build the documentation.
        Metadata (read only)
          This allows Read the Docs to read the repository collaborators and the permissions they have on the repository.
          This is used to determine if the user can connect a repository to a Read the Docs project.
        Pull requests (read and write)
          This allows Read the Docs to subscribe to pull request events,
          and to create a comment on the pull request with information about the build.

      Organization permissions
        Members (read only)
          This allows Read the Docs to read the organization members.

   .. tab:: GitHub (old OAuth app integration)

      .. note::

         Read the Docs used to use a GitHub OAuth application for integration,

>          which has been replaced by a `GitHub App <https://docs.github.com/en/apps/overview>`__.

Reviewer: ```suggestion
         which is being deprecated and replaced by a `GitHub App <https://docs.github.com/en/apps/overview>`__.
```

         If you haven't migrated your projects to the new GitHub App,

>          we will still use the OAuth application to interact with your repositories,

Reviewer: ```suggestion
         we will still use the OAuth application to interact with your repositories for now,
```
